Sources/BuildSystemIntegration/ExternalBuildSystemAdapter.swift

```diff
@@ -94,10 +94,34 @@ actor ExternalBuildSystemAdapter {
   private var lastRestart: Date?
 
   static package func projectRoot(for workspaceFolder: AbsolutePath, options: SourceKitLSPOptions) -> AbsolutePath? {
-    guard localFileSystem.isFile(workspaceFolder.appending(component: "buildServer.json")) else {
-      return nil
+    var buildServerConfigLocations = [String]()
+
+    #if os(Windows)
+    let workspacePath = workspaceFolder.pathString + "\\.bsp"
+    let localAppDataPath = ProcessInfo.processInfo.environment["LOCALAPPDATA"] ?? "C:\\Users\\Default\\AppData\\Local\\bsp"
```

Comment:
We shouldn’t hardcode any paths. I doubt that a user is named `Default` anyway. If the environment variable doesn’t exist, we shouldn’t search it. Same for the other missing environment variable below.